network/test/middleware_test.go

```diff
@@ -174,6 +182,200 @@ func (m *MiddlewareTestSuite) TestUpdateNodeAddresses() {
 	require.NoError(m.T(), err)
 }
 
+func (m *MiddlewareTestSuite) TestUnicastRateLimit_Messages() {
+	// limiter limit will be set to 5 events/sec the 6th event per interval will be rate limited
+	limit := rate.Limit(5)
+
+	// burst per interval
+	burst := 5
+
+	// create test time
+	testtime := unittest.NewTestTime()
+
+	// setup message rate limiter
+	messageRateLimiter := unicast.NewMessageRateLimiter(limit, burst, 1, unicast.WithGetTimeNowFunc(testtime.Now))
+
+	// the onUnicastRateLimitedPeerFunc call back we will use to keep track of how many times a rate limit happens
+	// after 5 rate limits we will close ch. O
+	ch := make(chan struct{})
+	var rateLimits uint64
```

Comment:
nit: use the `atomic.Uint64` class instead
```golang
rateLimits := atomic.NewUint64(0)
...
rateLimits.Inc()
```

Reply:
https://github.com/onflow/flow-go/pull/2840/commits/5051137f3bed50b8d269178ef5389a057a75f3df